crates/bevy_render/src/shader/shader_reflect.rs

```diff
@@ -33,6 +33,7 @@ impl ShaderLayout {
                 for input_variable in module.enumerate_input_variables(None).unwrap() {
                     if input_variable.name == GL_VERTEX_INDEX
                         || input_variable.name == GL_INSTANCE_INDEX
+                        || input_variable.name == GL_FRONT_FACING
                     {
                         continue;
                     }
```

Comment:
I might misunderstand the point of this condition, if so I'd love to learn. But wouldn't a more long-term solution be (snipped some `use` changes for clarity):

```diff
diff --git a/crates/bevy_render/src/shader/shader_reflect.rs b/crates/bevy_render/src/shader/shader_reflect.rs
index 238da1ef..95761fed 100644
--- a/crates/bevy_render/src/shader/shader_reflect.rs
+++ b/crates/bevy_render/src/shader/shader_reflect.rs
@@ -31,8 +32,9 @@ impl ShaderLayout {
                 // obtain attribute descriptors from reflection
                 let mut vertex_attributes = Vec::new();
                 for input_variable in module.enumerate_input_variables(None).unwrap() {
-                    if input_variable.name == GL_VERTEX_INDEX
-                        || input_variable.name == GL_INSTANCE_INDEX
+                    if input_variable
+                        .decoration_flags
+                        .contains(ReflectDecorationFlags::BUILT_IN)
                     {
                         continue;
                     }
```

Reply:
Oooh that does seem like a win / appears to do what we want it to.